src/integrated/soil_snow_model.jl

```diff
@@ -1,281 +1,435 @@
+        snow::SnM
+        "The soil model to be used"
+        soil::SoM
+    end
+
+A concrete type of land model used for simulating systems with
+snow and soil (and eventually rivers).
+$(DocStringExtensions.FIELDS)
+"""
+struct LandHydrologyModel{
+    FT,
+    SnM <: Snow.SnowModel{FT},
+    SoM <: Soil.EnergyHydrology{FT},
+} <: AbstractLandModel{FT}
+    "The snow model to be used"
+    snow::SnM
+    "The soil model to be used"
+    soil::SoM
+end
+
+"""
+    LandHydrologyModel{FT}(;
+        land_args::NamedTuple = (;),
+        snow_model_type::Type{SnM},
+        snow_args::NamedTuple = (;),
+        soil_model_type::Type{SoM},
+        soil_args::NamedTuple = (;),
+        ) where {
+            FT,
+            SnM <: Snow.SnowModel{FT},
+            SoM <: Soil.EnergyHydrology{FT},
+            }
+
+A constructor for the `LandHydrology`, which takes in the concrete model
+type and required arguments for each component, constructs those models,
+and constructs the `LandHydrologyModel` from them.
+
+Each component model is constructed with everything it needs to be stepped
+forward in time, including boundary conditions, source terms, and interaction
+terms.
+"""
+function LandHydrologyModel{FT}(;
+    land_args::NamedTuple = (;),
+    snow_model_type::Type{SnM},
+    snow_args::NamedTuple = (;),
+    soil_model_type::Type{SoM},
+    soil_args::NamedTuple = (;),
+) where {FT, SnM <: Snow.SnowModel, SoM <: Soil.EnergyHydrology}
+    (; atmos, radiation, domain) = land_args
+    prognostic_land_components = (:snow, :soil)
+    if :runoff ∈ propertynames(land_args)
+        top_bc = ClimaLand.AtmosDrivenFluxBC(
+            atmos,
+            radiation,
+            land_args.runoff,
+            prognostic_land_components,
+        )
+    else #no runoff model
+        top_bc = AtmosDrivenFluxBC(
+            atmos,
+            radiation,
+            ClimaLand.Soil.Runoff.NoRunoff(),
+            prognostic_land_components,
+        )
+    end
+    sources = (Soil.PhaseChange{FT}(),)
+    zero_flux = Soil.HeatFluxBC((p, t) -> 0.0)
+    boundary_conditions = (;
+        top = top_bc,
+        bottom = Soil.WaterHeatBC(;
+            water = Soil.FreeDrainage(),
+            heat = zero_flux,
+        ),
+    )
+    soil = soil_model_type{FT}(;
+        boundary_conditions = boundary_conditions,
+        sources = sources,
+        domain = domain,
+        soil_args...,
+    )
+    snow = snow_model_type(;
+        boundary_conditions = Snow.AtmosDrivenSnowBC(
+            atmos,
+            radiation,
+            prognostic_land_components,
+        ),
+        domain = Domains.obtain_surface_domain(domain),
+        snow_args...,
+    )
+
+    return LandHydrologyModel{FT, typeof(snow), typeof(soil)}(snow, soil)
+end
+
+"""
+    lsm_aux_vars(m::LandHydrologyModel)
+
+The names of the additional auxiliary variables that are
+included in the integrated Soil-Snow model.
+"""
+lsm_aux_vars(m::LandHydrologyModel) = (

```

Comment:
just curious - does the order of these matter? do snow aux vars come first and then soil (or vice versa), or are they intermixed?

Reply:
The order of them in the cache does not matter (I think we create the cache vars by first doing land components, and then adding in the lsm_aux_vars). But the order of how we update them does for sure matter.